res/res_sorcery_memory_cache.c

```diff
@@ -1507,8 +1507,9 @@ static int age_cmp(void *a, void *b)
  */
 static void *sorcery_memory_cache_open(const char *data)
 {
-	char *options = ast_strdup(data), *option;
```

Comment:
I don't see why this can't just use strdupa, other modules do that.

Reply:
Initially, I intended to switch to strdupa. However, since it was unclear whether the original use of strdup was due to a spelling mistake or to account for potential long input data, I decided to add resource deallocation based on that. Of course, using strdupa would have been a simpler and more efficient option.

Reply:
Other implementations do it, and this isn't a loop where it will continually allocate, so it's safe.